vectordb_bench/backend/clients/oceanbase/oceanbase.py

```diff
@@ -0,0 +1,215 @@
+import logging
+import time
+from contextlib import contextmanager
+from typing import Generator, Any, Tuple, Optional, List, Dict
+
+import mysql.connector as mysql
+import numpy as np
+from ..api import VectorDB, IndexType, MetricType
+from .config import OceanBaseIndexConfig, OceanBaseConfigDict
+import struct
+
+log = logging.getLogger(__name__)
+
+OCEANBASE_DEFAULT_LOAD_BATCH_SIZE = 256
+
+
+class OceanBase(VectorDB):
+    def __init__(
+        self,
+        dim: int,
+        db_config: OceanBaseConfigDict,
+        db_case_config: OceanBaseIndexConfig,
+        collection_name: str = "items",
+        drop_old: bool = False,
+        **kwargs,
+    ):
+        self.name = "OceanBase"
+        self.dim = dim
+        self.db_config = db_config
+        self.db_case_config = db_case_config
+        self.table_name = collection_name
+        self.load_batch_size = OCEANBASE_DEFAULT_LOAD_BATCH_SIZE
+        self._index_name = "vidx"
+        self._primary_field = "id"
+        self._vector_field = "embedding"
+        if self.db_case_config.index == IndexType.HNSW_BQ:
+            self.db_case_config.metric_type = MetricType.L2
```

Comment:
It seems that `HNSW_BQ` only supports the `L2` metric. We suggest that the metric conversion logic could be placed within the `parse_metric()` method of the `OceanBaseHNSWConfig` class.

Additionally, we have provided a `need_normalize_cosine` method. You can return true or false based on the actual implementation. If it returns true, vdbbench will **normalize the original data** for both insertion and querying. This approach ensures the correctness of the ground truth, which is used for calculating the standard recall, for databases that do not natively support cosine similarity in some cases.